### PR TITLE
Feature/add rabbit mq to service

### PR DESCRIPTION
**Pull request by jschwait**
Added unit tests using jasmine

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -2,6 +2,8 @@ FROM node:carbon
 WORKDIR /src/
 COPY package*.json ./
 RUN npm install
+RUN npm install amqplib
+RUN npm install xmlhttprequest
```
Comment (dmssargent) [inline Dockerfile]: Are these packages not in package.json?

File: src/utils/rabbitmq-utils.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+const XMLHttpRequest = require('xmlhttprequest').XMLHttpRequest
+
+function sleep (ms) {
+  return new Promise(resolve => setTimeout(resolve, ms))
+}
+
+function rabbitMQHealthCheck (rabbitHost) {
+  var request = new XMLHttpRequest()
+  request.withCredentials = true
+  request.open('GET', 'http://' + rabbitHost + ':15672/api/healthchecks/node', false, 'guest', 'guest')
```
Comment (dmssargent) [inline src/utils/rabbitmq-utils.js:11]: What happened to `process.env.RABBIT_PASSWORD || 'guest'` and the likewise?

File: src/utils/rabbitmq-consumer.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  return generatedConsumerFunction
+}
+
+function initializeConsumer (exchangeName, exchangeKind, queueName, queueBinding, consumerFunction) {
+  var rabbitUsername = process.env.RABBIT_USERNAME || 'guest'
+  var rabbitPassword = process.env.RABBIT_PASSWORD || 'guest'
+  var rabbitHost = process.env.RABBIT_HOST || 'localhost'
+
+  winston.log('info', 'before dialing rabbitmq. Attempting to connect to: amqp://' + rabbitUsername + ':' + rabbitPassword + '@' + rabbitHost + ':5672/')
+  amqp.connect('amqp://' + rabbitUsername + ':' + rabbitPassword + '@' + rabbitHost + ':5672/', function (err, conn) {
```
Comment (dmssargent) [inline src/utils/rabbitmq-consumer.js]: Would string interpolation be helpful here?

File: src/utils/rabbitmq-utils.js (hunk 2 above)
Comment (jschwait) [reply src/utils/rabbitmq-utils.js]: no idea, I  only added the spec folder

File: Dockerfile (hunk 1 above)
Comment (jschwait) [reply Dockerfile]: no

Comment (dmssargent) [reply Dockerfile]: If you add them to package.json, do you still need to manually install them?

File: spec/rabbitmqHandlersSpec.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
+const rabbitMqHandlers = require('../src/handlers/rabbitmq-handlers')
+// Test valid rabbitmq-handlers input
+describe('rabbitmq-handlers', () => {
+  // it('should have valid content', (done) => {
```
Comment (jusoto) [inline spec/rabbitmqHandlersSpec.js]: If the commented code is not going to be used, it should be deleted

File: src/utils/rabbitmq-utils.js (hunk 2 above)
Comment (jschwait) [reply src/utils/rabbitmq-utils.js:11]: This function is called in 'waitForRabbitMQ'.  The parameter passed in is: process.env.RABBIT_PASSWORD || 'guest'

So it's still there, just refactored

File: spec/rabbitmqHandlersSpec.js (hunk 4 above)
Comment (jschwait) [reply spec/rabbitmqHandlersSpec.js]: Fixed

File: src/utils/rabbitmq-utils.js (hunk 2 above)
Comment (dmssargent) [reply src/utils/rabbitmq-utils.js:11]: Gotcha

File: package.json [hunk 5]
```diff
@@ -19,15 +19,25 @@
...
   "dependencies": {
     "@google/maps": "^0.4.5",
+    "amqp": "^0.2.7",
+    "amqplib": "^0.5.2",
     "eureka-js-client": "^4.4.1",
+    "fix": "0.0.6",
     "ip": "^1.1.5",
-    "npm": "^5.7.1",
+    "jasmine": "^3.1.0",
+    "npm": "^6.2.0",
```
Comment (dmssargent) [inline package.json]: I don't think you need to install npm per package, if you need to specify a version of npm it should use the engines property: 
https://docs.npmjs.com/files/package.json#engines

**Review by dmssargent — CHANGES_REQUESTED** [review]

**Review by dmssargent — APPROVED** [review]

**Review by jusoto — APPROVED** [review]
Looks good

**Pull request merged**